### PR TITLE
Ascent in situ integration: add support for particles

### PR DESCRIPTION
**Pull request by cyrush**
This PR adds support for wrapping WrapX particle containers so they can be published and rendered in Ascent. 

With this change Ascent can render both particles and the AMReX mesh data. 

I updated the Ascent WarpX docs with an example yaml input that shows how to render both an isosurface of the mesh and a particle field. 

This PR requires AMReX develop that includes this PR https://github.com/AMReX-Codes/amrex/pull/734 (already merged) and a recent version of Ascent. 

~~One outstanding issue: The overall UpdateInSitu code path crashes in WarpX::AverageAndPackFields for my example. (see: https://github.com/ECP-WarpX/WarpX/issues/788)~~ -> #837

I am happy to help track this issue done with more guidance.

### CODE REVIEW

**Comment by dpgrote** [comment]
Thinking about this some more, I might suggest that the bug fix in AverageAndPackFields be in a separate PR since it will affect other things beyond the changes in this PR. I'd be happy to do that if you agree.

**Comment by cyrush** [comment]
@dpgrote thats fine with me -- I am open to what folks feel is the best path. 

I can remove that change from my branch - or we can wait to merge this change until that goes in, etc

**Comment by ax3l** [comment]
@cyrush P.S.: please don't forget to rebase this PR as soon as #837 is merged :)

**Comment by ax3l** [comment]
@cyrush ready for rebase against `master` now :)

**Comment by cyrush** [comment]
@ax3l -- tackled the rebase, I'll work on the docs suggestions a bit later.

File: Docs/source/visualization/ascent.rst [hunk 1]
```diff
@@ -45,13 +45,14 @@ The supported parameters are described in the following table.
 +-------------------------+------------------------------------------------------+---------+
 
 A typical use case is setting :code:`insitu.int` to a value of one or greater and
-:code:`insitu.start` to the first time step where in situ analyswhere in situ analysis should be
+:code:`insitu.start` to the first time step where in situ analysis should be
```
Comment (ax3l) [inline Docs/source/visualization/ascent.rst]: We also document those params in `Docs/source/running_cpp/parameters.rst`

Comment (ax3l) [reply Docs/source/visualization/ascent.rst]: @lge0303 maybe this can be added a follow-up PR of yours? :)

**Comment by lge0303** [comment]
Yes 😊

**Comment by cyrush** [comment]
@ax3l thanks for getting this in!

**Comment by ax3l** [comment]
Thank you all for your awesome contribution! :rocket: :sparkles:

**Pull request merged**